crates/pipeline-manager/src/api/endpoints/cluster_healthz.rs

```diff
@@ -0,0 +1,27 @@
+use actix_web::{get, web::Data as WebData, HttpResponse, Responder};
+
+use crate::api::main::ServerState;
+
+#[utoipa::path(
+    get,
+    path = "/cluster_healthz",
+    responses(
+        (status = 200, description = "All services healthy", body = HealthStatus),
+        (status = 503, description = "One or more services unhealthy", body = HealthStatus)
+    )
+)]
+#[get("/cluster_healthz")]
+pub(crate) async fn get_health(state: WebData<ServerState>) -> impl Responder {
```

Comment:
this doesn't need auth? is it expected?

Reply:
I think putting it behind auth sounds good